src/sagas/messages/sendMessage/sendMessage.saga.test.ts

```diff
@@ -12,94 +12,139 @@ import { Socket } from 'socket.io-client';
 import { StoreKeys } from '../../store.keys';
-import {
-  messagesActions,
-  messagesReducer,
-  MessagesState,
-} from '../messages.slice';
+import { messagesActions } from '../messages.slice';
 import { sendMessageSaga } from './sendMessage.saga';
-
-import {
-  publicChannelsReducer,
-  PublicChannelsState,
-} from '../../publicChannels/publicChannels.slice';
 import { SocketActionTypes } from '../../socket/const/actionTypes';
 import { MessageTypes } from '../const/messageTypes';
 import { generateMessageId, getCurrentTime } from '../utils/message.utils';
 import { Identity } from '../../identity/identity.slice';
 import { identityAdapter } from '../../identity/identity.adapter';
+import {
+  communitiesReducer,
+  CommunitiesState,
+  Community,
+} from '../../communities/communities.slice';
+import { communitiesAdapter } from '../../communities/communities.adapter';
+import {
+  CommunityChannels,
+  publicChannelsReducer,
+  PublicChannelsState,
+} from '../../publicChannels/publicChannels.slice';
+import {
+  channelsByCommunityAdapter,
+  publicChannelsAdapter,
+} from '../../publicChannels/publicChannels.adapter';
+import { IChannelInfo } from 'src';
 
 describe('sendMessageSaga', () => {
+  const communityId = 'id';
+
+  const community = new Community({
+    id: communityId,
+    name: 'community',
+    CA: 'CA',
+    registrarUrl: 'registrarUrl',
+  });
+
   const identity = new Identity({
-    id: 'id',
+    id: communityId,
     hiddenService: { onionAddress: 'onionAddress', privateKey: 'privateKey' },
     dmKeys: { publicKey: 'publicKey', privateKey: 'privateKey' },
-    peerId: { id: 'peerId', pubKey: 'pubKey', privKey: 'privKey' },
+    peerId: { id: 'id', pubKey: 'pubKey', privKey: 'privKey' },
   });
 
-  test.skip('sign and send message', async () => {
-    const socket = { emit: jest.fn() } as unknown as Socket;
-    const csr = {
-      userCsr: 'userCsr',
-      userKey: 'userKey',
+  const csr = {
+    userCsr: 'userCsr',
+    userKey: 'userKey',
+    pkcs10: {
+      publicKey: jest.fn() as unknown as KeyObject,
+      privateKey: jest.fn() as unknown as KeyObject,
       pkcs10: {
-        publicKey: jest.fn() as unknown as KeyObject,
-        privateKey: jest.fn() as unknown as KeyObject,
-        pkcs10: {
-          userKey: jest.fn() as unknown,
-        },
+        userKey: jest.fn() as unknown,
       },
-    };
+    },
+  };
+
+  identity.userCertificate = 'userCertificate'
+  identity.userCsr = csr
+
+  const publicChannel: IChannelInfo = {
+    name: 'general',
+    description: 'description',
+    owner: 'user',
+    timestamp: 0,
+    address: 'address',
+  };
+
+  const communityChannels = new CommunityChannels(communityId);
+  communityChannels.currentChannel = publicChannel.address
+  communityChannels.channels = publicChannelsAdapter.setAll(
+    publicChannelsAdapter.getInitialState(),
+    [publicChannel]
+  )
+
+  test('sign and send message', async () => {
```

Comment:
Is this test actually checking anything, or just making sure that everything runs without an error? 

Is this the place to have a test that checks that the message is signed using the correct key, added to the correct channel, has the correct structure, etc?

Reply:
Yes, saga which we test here is quite complex so the test checks if send message action will dispatch. Also all the things you mentioned are included (except we mock identity methods but that's ok as nectar's tests has nothing to do with it)